addons/mail/static/src/models/message_action.js

```diff
@@ -0,0 +1,147 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, one } from '@mail/model/model_field';
+import { clear, insertAndReplace } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'MessageAction',
+    identifyingMode: 'xor',
+    recordMethods: {
+        /**
+         * @private
+         * @returns {integer}
+         */
+        _computeIsNonCompactActionContribution() {
+            if (this.messageActionListOwnerAsToggleCompact) {
+                return 0;
+            }
+            return 1;
+        },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeMessageActionListOwner() {
+            if (this.messageActionListOwnerAsDelete) {
+                return this.messageActionListOwnerAsDelete;
+            } else if (this.messageActionListOwnerAsEdit) {
+                return this.messageActionListOwnerAsEdit;
+            } else if (this.messageActionListOwnerAsMarkAsRead) {
+                return this.messageActionListOwnerAsMarkAsRead;
+            } else if (this.messageActionListOwnerAsReaction) {
+                return this.messageActionListOwnerAsReaction;
+            } else if (this.messageActionListOwnerAsReplyTo) {
+                return this.messageActionListOwnerAsReplyTo;
+            } else if (this.messageActionListOwnerAsToggleCompact) {
+                return this.messageActionListOwnerAsToggleCompact;
+            } else if (this.messageActionListOwnerAsToggleStar) {
+                return this.messageActionListOwnerAsToggleStar;
+            }
+            return clear();
+        },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeMessageActionView() {
+            if (
+                this.messageActionListOwner &&
+                !this.messageActionListOwner.actionToggleCompact ||
+                this.messageActionListOwner.actionToggleCompact &&
+                !this.messageActionListOwner.isCompact
+            ) {
+                return insertAndReplace();
+            } else if (
+                this.messageActionListOwner &&
+                this.messageActionListOwner.actionToggleCompact &&
+                this.messageActionListOwner.isCompact
+            ) {
+                if (
+                    this.messageActionListOwner == this.messageActionListOwnerAsReaction ||
+                    this.messageActionListOwner == this.messageActionListOwnerAsToggleCompact
```

Comment:
`===`, but isn't just checking the existence of `messageActionListOwnerAsReaction` or `messageActionListOwnerAsToggleCompact` enough?